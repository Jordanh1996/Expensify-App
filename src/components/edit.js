import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './expenseForm';
import {editExpense, removeExpense} from '../actions/expenses';

export class Editpage extends React.Component {

    onSubmit = (expense) => {
        this.props.editExpense(this.props.match.params.id, expense)
        this.props.history.push('/')
    }

    removeExpense = () => {
        this.props.removeExpense({id: this.props.expense.id})
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.removeExpense}
            >
                        remove
                </button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => {dispatch(editExpense(id, expense))},
    removeExpense: (data) => {dispatch(removeExpense(data))}
})


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Editpage);