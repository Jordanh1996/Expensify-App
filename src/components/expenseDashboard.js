import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilters from './expenseListFilters';

const ExpenseDashboard = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDashboard;