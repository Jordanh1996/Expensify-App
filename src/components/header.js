import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/' activeClassName="is-active" exact={true}>this is dashboard </NavLink>
        <NavLink to='/edit' activeClassName="is-active">this is edit </NavLink>
        <NavLink to='/help' activeClassName="is-active">this is help </NavLink>
        <NavLink to='/expense' activeClassName="is-active">add expense</NavLink>
    </header>
)

export default Header;