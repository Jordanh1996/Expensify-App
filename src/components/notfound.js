import React from 'react';
import {Link} from 'react-router-dom';

const Notfound = () => (
    <div>
        <Link to='/'>404</Link>
    </div>
)

export default Notfound;