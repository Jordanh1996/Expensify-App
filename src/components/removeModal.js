import React from 'react';
import Modal from 'react-modal';

const RemoveModal = (props) => (
    
    <Modal
    isOpen={props.isopen}
    contentLabel="Remove Expense"
    onRequestClose={props.isclose}
    closeTimeoutMS={200}
    className="modal"
    >

    <p>Are you sure you want to remove this option?</p>
    <div className="button--modal">
        <button onClick={props.confirmRemove} className="button">Confirm</button>
        <button onClick={props.isclose} className="button button--secondary">Cancel</button>
    </div>
    </Modal>

)

export default RemoveModal;