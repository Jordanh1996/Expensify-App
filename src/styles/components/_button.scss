
.button {
    color: white;
    background-color: $blue;
    padding: $s-size;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    border: none;
    line-height: 1;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}