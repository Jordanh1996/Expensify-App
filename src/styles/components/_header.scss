.header {
    background: $dark-blue;

}

.header__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
}

.header__title {
    color: white;
    text-decoration: none;

    h1 {
        margin: 0;
    }
}

.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
    span {
        font-weight: bold;
    }
}

.page-header__title {
    margin: 0;
    font-weight: 300;
}

.page-header__actions {
    margin-top: $m-size;
    
}