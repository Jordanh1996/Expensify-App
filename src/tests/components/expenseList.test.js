import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/expenseList';
import expenses from '../fixture/expenses';

test('should render expenseList with expenses props correctly', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render expenseList with the paragraph tag', () => {
    const wrapper = shallow(<ExpenseList expenses = {[]} />)
    expect(wrapper).toMatchSnapshot()
})