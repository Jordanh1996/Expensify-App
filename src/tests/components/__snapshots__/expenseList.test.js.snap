// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList with expenses props correctly 1`] = `
<div>
  <ExpenseItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render expenseList with the paragraph tag 1`] = `
<div>
  <p>
    there are no expenses
  </p>
</div>
`;
